libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -81,16 +82,21 @@ export function OutQueueManager(
   customHeaders: Record<string, string>,
   withCredentials: boolean,
   retryStatusCodes: number[],
-  dontRetryStatusCodes: number[]
+  dontRetryStatusCodes: number[],
+  idService?: string
 ): OutQueue {
   type PostEvent = {
     evt: Record<string, unknown>;
     bytes: number;
   };
 
+  type IdServiceCall = {
+    url: string;
+  };
+
   let executingQueue = false,
     configCollectorUrl: string,
-    outQueue: Array<PostEvent> | Array<string> = [];
+    outQueue: Array<string | PostEvent | IdServiceCall> = [];
```

Comment:
I think this type change hides some assumptions other parts of the code make, per the below.

`PostEvent[] | string[]` > `(PostEvent | string | IdServiceCall)[]`

We might be better off just having a flag for if we've contacted the idService yet, and when we start processing the queue, and handle it if it's not set?

Reply:
I agree that having a flag would make the implementation more straightforward. I think it wouldn't change the number of if conditions we have to make and would make the type of the event queue simpler.